Descent3/init.cpp

```diff
@@ -1406,16 +1404,30 @@ void InitIOSystems(bool editor) {
       Error("Failed to get executable path\n");
     } else {
       std::filesystem::path executablePath(exec_path);
-      std::string baseDirectoryString = executablePath.parent_path().string();
-      strncpy(Base_directory, baseDirectoryString.c_str(), sizeof(Base_directory) - 1);
-      Base_directory[sizeof(Base_directory) - 1] = '\0';
-      LOG_INFO << "Using working directory of " << Base_directory;
+      writable_base_directory = executablePath.parent_path();
+      LOG_INFO << "Using working directory of " << writable_base_directory.u8string().c_str();
     }
   } else {
-    ddio_GetWorkingDir(Base_directory, sizeof(Base_directory));
+    writable_base_directory = std::filesystem::current_path();
+  }
+
+  ddio_SetWorkingDir(writable_base_directory.u8string().c_str());
+  initial_base_directories.insert(initial_base_directories.begin(), writable_base_directory);
+
+  // Set any additional base directories
+  auto additionaldirarg = 0;
+  while (0 != (additionaldirarg = FindArg("-additionaldir", additionaldirarg))) {
+    const auto dir_to_add = GetArg(additionaldirarg + 1);
+    if (dir_to_add == NULL) {
+      LOG_WARNING << "-additionaldir was at the end of the argument list. It should never be at the end of the argument list.";
```

Comment:
Probably this should means that user passed `-additionaldir` without a value?

Reply:
So, you’re saying that it would be better if the error message said something along the lives of “There was no value for the final -additionaldir option”?